autotest/autotest.cpp

```diff
@@ -557,10 +557,18 @@ class pcap_expectation
 	std::vector<u_char> buffer;
 };
 
+static std::string
+generate_temp_filepath(const std::filesystem::path& dir = std::filesystem::temp_directory_path())
+{
+	auto time = std::chrono::system_clock::now().time_since_epoch().count();
+	std::string filename = "file" + std::to_string(time);
+	return dir / filename;
+}
+
 void tAutotest::recvThread(std::string interfaceName,
                            std::vector<std::string> expectFilePaths)
 {
-	PcapDumper pcapDumper(std::tmpnam(nullptr) + std::string(".pcap"));
+	PcapDumper pcapDumper(generate_temp_filepath() + ".pcap");
```

Comment:
I would prefer to use `mkstemp` here because we may to run test in parallel and the code above is not concurrent safe.